### PR TITLE
SDR fixes for Release mode compile.

### PR DESCRIPTION
**Pull request by ctrl-z-9000-times**
Per @dkeeney  request in issue #167 
Thanks for pointing problem out!

### CODE REVIEW

File: src/test/CMakeLists.txt [hunk 1]
```diff
@@ -92,6 +92,7 @@ set(ntypes_tests
 	   unit/ntypes/MemStreamTest.cpp
 	   unit/ntypes/NodeSetTest.cpp
 	   unit/ntypes/ScalarTest.cpp
+	   unit/ntypes/SdrTest.cpp
```
Comment (breznak) [inline src/test/CMakeLists.txt]: this is a good one, SDR test were not run? :)

**Comment by ctrl-z-9000-times** [comment]
> this might be applicable to more places in the codebase

We should know soon if there are more places with for-loop{assert} because it causes an error when doing a release build, since the for loop iterator is unused in an empty loop.

**Comment by breznak** [comment]
> it causes an error when doing a release build 

we run Release on Travis Linux, and it is always green. Might be a spot by some compiler/OS combination (David on OSX...)

**Comment by dkeeney** [comment]
For some reason, OSX catches more unused variable errors than travis.

**Comment by breznak** [comment]
Merged, Linux passes, OSX is at the same `undefined symbol error` #167

**Pull request merged**